src/protocol/mul/semi_honest.rs

```diff
@@ -103,40 +100,46 @@ pub mod tests {
     #[tokio::test]
     #[allow(clippy::cast_possible_truncation)]
     pub async fn concurrent_mul() {
-        type MulArgs<F> = (Replicated<F>, Replicated<F>);
-        async fn mul<F: Field>(
-            v: (ProtocolContext<'_, Replicated<F>, F>, MulArgs<F>),
-        ) -> Replicated<F> {
-            let (ctx, (a, b)) = v;
-            ctx.multiply(RecordId::from(0), &a, &b).await.unwrap()
-        }
-
         let world = make_world(QueryId);
         let contexts = make_contexts::<Fp31>(&world);
-        let mut rand = StepRng::new(1, 1);
-
-        let mut multiplications = Vec::new();
-
-        for step in 1..10_u8 {
-            let a = share(Fp31::from(4_u128), &mut rand);
-            let b = share(Fp31::from(3_u128), &mut rand);
-
-            let step_name = format!("step{}", step);
-            let f = join_all(
-                contexts
-                    .iter()
-                    .map(|ctx| ctx.narrow(&step_name))
-                    .zip(zip(a, b))
-                    .map(mul),
-            );
-            multiplications.push(f);
-        }
-
-        let results = join_all(multiplications).await;
-        for shares in results {
+        let mut rng = rand::thread_rng();
+
+        let mut expected_outputs = Vec::with_capacity(10);
+
+        let futures: Vec<_> = zip(repeat(contexts), 0..10)
+            .map(|(ctx, i)| {
+                let a = rng.gen::<Fp31>();
+                let b = rng.gen::<Fp31>();
+                expected_outputs.push(a * b);
+
+                let a_shares: [Replicated<Fp31>; 3] = share(a, &mut rng);
+                let b_shares: [Replicated<Fp31>; 3] = share(b, &mut rng);
```

Comment:
```suggestion
                let a_shares = share(a, &mut rng);
                let b_shares = share(b, &mut rng);
```

there may be a reason why it wouldn't work but I don't see any - `a` and `b` are `Fp31`